h2o-algos/src/main/java/hex/tree/dt/DT.java

```diff
@@ -0,0 +1,399 @@
+                        .mapToObj(i -> data.vec(i))
+                        // decrease min as the minimum border is always excluded and real min value could be lost
+                        .map(v -> new FeatureLimits(v.min() - EPSILON, v.max()))
+                        .collect(Collectors.toList()));
+    }
+
+
+    private class DTDriver extends Driver {
+
+        private void dtChecks() {
+            if (_train.hasNAs()) {
+                error("_train", "NaNs are not supported yet");
+            }
+            if (_train.hasInfs()) {
+                error("_train", "Infs are not supported");
+            }
+            if (IntStream.range(0, _train.numCols() - 1) // ignore prediction column
+                    .mapToObj(index -> _train.vec(index).isCategorical()).anyMatch(i -> i)) {
+                error("_train", "Categorical features are not supported yet");
+            }
+            if (!_response.isCategorical()) {
+                error("_response", "Only categorical response is supported");
+            }
+            if (!_response.isBinary()) {
+                error("_response", "Only binary response is supported");
+            }
+        }
+
+        @Override
+        public void computeImpl() {
+            _model = null;
+            try {
+                init(true);
+                dtChecks();
+                if (error_count() > 0) {
+                    throw H2OModelBuilderIllegalArgumentException.makeFromBuilder(DT.this);
+                }
+                _rand = RandomUtils.getRNG(_parms._seed);
+                _model = new DTModel(dest(), _parms,
+                        new DTModel.DTOutput(DT.this));
+                _model.delete_and_lock(_job);
+                buildDT();
+                LOG.info(_model.toString());
+            } finally {
+                if (_model != null)
+                    _model.unlock(_job);
+            }
+        }
+
+        /**
+         * Build SDT and update infrastructure.
+         */
+        private void buildDT() {
+            buildDTIteratively();
+            Log.debug("depth: " + _parms._max_depth + ", nodes count: " + _nodesCount);
+
+            CompressedDT compressedDT = new CompressedDT(_tree);
+
+            _model._output._treeKey = compressedDT._key;
+            DKV.put(compressedDT);
+            _job.update(1);
+            _model.update(_job);
+//            System.out.println("Tree: " + compressedDT.toString());
+//            System.out.println("Rules: " + String.join("\n", compressedDT.getListOfRules()));
+            Log.debug("Tree:");
+            Log.debug(Arrays.deepToString(_tree));
+        }
+
+        /**
+         * Build the tree iteratively starting from the root node.
+         */
+        private void buildDTIteratively() {
+            _tree = new double[(int) Math.pow(2, _parms._max_depth + 1)][3];
+            Queue<DataFeaturesLimits> limitsQueue = new LinkedList<>();
```

Comment:
Number of nodes for a tree with max_depth is

Math.pow(2, _parms._max_depth+1)-1